cortex_serving_client/cortex_client.py

```diff
@@ -529,10 +531,22 @@ def _start_gc_loop(self, interval_sec):
 @contextmanager
 def open_pg_cursor(db_connection_pool, key=None):
     try:
-        with db_connection_pool.getconn(key) as conn:
-            conn.autocommit = True
-            with conn.cursor() as cur:
-                yield cur
+        for retry in range(DB_TRIES):
+            try:
+                with db_connection_pool.getconn(key) as conn:
+                    conn.autocommit = True
+                    with conn.cursor() as cur:
+                        yield cur
+
+                    break
+
+            except PoolError as e:
+                if retry + 1 >= DB_TRIES:
```

Comment:
why not:
`if retry > DB_TRIES:`

Reply:
`retry` can reach only to `DB_TRIES - 1`. At the same time if the for loop ends it must have been because no exception was thrown.